third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/MFMA.cpp

```diff
@@ -278,11 +278,18 @@ struct DotOpMFMAConversionHelper {
     int kpack = kWidth / kBase;
     SmallVector<Value> results;
     auto vecTy = vec_ty(type, kBase);
+    if (type.isBF16())
+      vecTy = vec_ty(i16_ty, kBase);
```

Comment:
It should be simpler if we still pass i16_ty to extractOperands if type is bf16. Therefore, we won't need this check.

Reply:
> It should be simpler if we still pass i16_ty to extractOperands if type is bf16. Therefore, we won't need this check.

But `extract_element(type, rawElems, i32_val(elemId + k * kBase))` is not compatible for `i16_ty` after remove the typeConverter rule which treat bf16 as i16.

Reply:
Ah I see, thank you for the explanation.